dolfyn/tools/misc.py

```diff
@@ -2,23 +2,27 @@
 from scipy.signal import medfilt2d, convolve2d
 
 
-def nans(*args, **kwargs):
+def _nans(*args, **kwargs):
     out = np.empty(*args, **kwargs)
-    out[:] = np.NaN
-    return out
+    if np.issubdtype(out.flatten()[0], np.integer):
+        out[:] = 0
+        return out
+    else:
+        out[:] = np.nan
+        return out
 
 
-def nans_like(*args, **kwargs):
+def _nans_like(*args, **kwargs):
     out = np.empty_like(*args, **kwargs)
-    out[:] = np.NaN
+    out[:] = np.nan
     return out
 
 
-def find(arr):
+def _find(arr):
     return np.nonzero(np.ravel(arr))[0]
 
 
-def detrend(arr, axis=-1, in_place=False):
```

Comment:
I think this should be included in the API.

Reply:
Done. I'll push once I've written some basic tests for these, and add something to the documentation.